csrc/fusion_segmenter.cpp

```diff
@@ -4467,75 +4639,73 @@ void SegmentCandidateFinder::revertPrivatizedUpcast(SegmentedGroup* group) {
     }
   };
 
-  for (const auto& [original_upcast, clones] : privatized_upcast_ops_) {
-    std::vector<UnaryOp*> upcast_in_group;
-    Val* upcast_val_to_keep = nullptr;
-    for (auto uop : ir_utils::filterByType<UnaryOp>(group->exprs())) {
-      if (uop != original_upcast && !clones.count(uop)) {
+  for (const auto& [original, clones] : privatized_ops_) {
```

Comment:
@naoyam , there are no iteration over reverting the privatized ops anymore. 
This is almost just a change in names of variables.